src/pepr/operator/controllers/network/policies.ts

```diff
@@ -59,36 +91,54 @@ export async function networkPolicies(pkg: UDSPackage, namespace: string, istioM
   const exposeList = pkg.spec?.network?.expose ?? [];
   // Iterate over each exposed service, excluding directResponse services
   for (const expose of exposeList.filter(exp => !exp.advancedHTTP?.directResponse)) {
-    const { gateway = Gateway.Tenant, port, selector = {}, targetPort } = expose;
-
-    // Use the same port as the VirtualService if targetPort is not set
+    const { gateway = Gateway.Tenant, port = 80, selector = {}, targetPort } = expose;
     const policyPort = targetPort ?? port;
 
+    // Find if this service has a matching client with waypoint
+    const matchingClient = findMatchingClient(pkg, selector);
+    const waypointName = matchingClient ? getWaypointName(matchingClient.clientId) : undefined;
+
+    // Use waypoint selector only if we have a waypoint and the package is configured for ambient waypoint
+    const podSelector =
+      waypointName && shouldUseAmbientWaypoint(pkg)
+        ? getPodSelector(pkg, selector, waypointName)
+        : selector;
```

Comment:
Doesn't `getPodSelector` already take into account `shouldUseAmbientWaypoint` and handle the logic of falling back to selector?